File: first.py
f = open("inputs/first.txt", "r")
lines = f.read().splitlines()

x = y = 0
for line in lines:
    tmpx = int(line)
    for line2 in lines:
        tmpy = int(line2)
        if (tmpx+tmpy) == 2020:
            x = tmpx
            y = tmpy
result = x * y

for line in lines:
    tmpx = int(line)
    for line2 in lines:
        tmpy = int(line2)
        for line3 in lines:
            tmpz = int(line3)
            if (tmpx+tmpy+tmpz) == 2020:
                x = tmpx
                y = tmpy
                z = tmpz

result = x*y*z
print(result)

File: first.pl
read_lines(Lines):-
  open('inputs/first.txt',read,Stream),
  read_file(Stream,Lines),
  close(Stream).

read_file(Stream,[]) :-
  at_end_of_stream(Stream).

read_file(Stream,[X|L]) :-
  \+ at_end_of_stream(Stream),
  read(Stream,X),
  read_file(Stream,L).


first_star(RESULT) :-
  read_lines(L),
  member(Y, L),
  member(X, L),
  2020 is X+Y,
  RESULT is X*Y.

second_star(RESULT) :-
  read_lines(L),
  member(Y, L),
  member(X, L),
  member(Z, L),
  2020 is X+Y+Z,
  RESULT is X*Y.
